source/main.cpp

```diff
@@ -355,30 +361,36 @@ std::string getInput(HB_Keyboard* sHBKB, bool &bCancelled)
         if (KBState == 3)
         {
             bCancelled = true;
+            input = "";
             break;
         }
         // Otherwise if the user has entered a key
         else if (KBState != 4)
         {
-            printf("%c[2K\r", 27);
-            printf("%s", input.c_str());
+            if(strcmp(last_input.c_str(),input.c_str()) != 0) {
+                ui_menu_draw_string(input.c_str(), 10, 10, COLOR_SELECTED);
+                sceneDraw();
+                last_input = input;
+            }
+            //Wait for VBlank
```

Comment:
We may need to do something about these. HBKB doesn't draw properly without them (Some buttons get stuck with a weird flickery blue color to them). Not sure how to handle it though, as with these two lines in, the top screen goes flickery while typing.